stl/inc/yvals_core.h

```diff
@@ -435,18 +435,42 @@
 #define _NODISCARD_FRIEND _NODISCARD friend
 #endif // TRANSITION, VSO-568006
 
-// Determine if we should use [[msvc::known_semantics]] to communicate to the compiler
-// that certain type trait specializations have the standard-mandated semantics
+#pragma push_macro("msvc")
+#pragma push_macro("known_semantics")
+#pragma push_macro("noop_dtor")
+#undef msvc
+#undef known_semantics
+#undef noop_dtor
+
 #ifndef __has_cpp_attribute
-#define _MSVC_KNOWN_SEMANTICS
+#define _HAS_MSVC_ATTRIBUTE(x) 0
 #elif defined(__CUDACC__) // TRANSITION, CUDA - warning: attribute namespace "msvc" is unrecognized
-#define _MSVC_KNOWN_SEMANTICS
-#elif __has_cpp_attribute(msvc::known_semantics)
+#define _HAS_MSVC_ATTRIBUTE(x) 0
+#else
+#define _HAS_MSVC_ATTRIBUTE(x) __has_cpp_attribute(msvc::x)

```

Comment:
No change requested: This is *much* more elegant than I envisioned, thank you! :heart_eyes_cat: